### ISSUE TITLE
Please add to Flatpak the Python engine

### ISSUE THREAD

**Issue by Alexey-T** [flathub/io.github.cudatext.CudaText-Qt5#2]
According to
https://github.com/Alexey-T/CudaText/issues/4357
the normal Python engine cannot be called from Flatpak version. So you need to add the Py engine to the Flatpak. It can be 3.8.x to 3.10.x.

**Comment by Alexey-T** [comment]
You also need to preconfigure CudaText to use this new Py engine! to do this - create the user.json in the app "settings" dir (usual localtion - https://wiki.freepascal.org/CudaText#Location_of_folders_.27settings.27.2C_.27py.27.2C_.27data.27 )

it must have the like like this:

```
{
  "pylib__linux" : "/usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0",
}
```

**Comment by hfiguiere** [comment]
Python 3.9 is in the runtime.

**Comment by Alexey-T** [comment]
> Python 3.9 is in the runtime.

then what is the full path to libpython3*.so file for the CudaText?

**Comment by hfiguiere** [comment]
it's in the library path. It should just find it.

**Comment by bogen85** [comment]
@hfiguiere @Alexey-T Alright, the issue is not the finding of python. That (not the system wide python) is found and most plugins work fine. (What is found is a python that flatpak installs for the io.github.cudatext.CudaText-Qt5 package, which, as I said, works fine).

The issue is when a CudaText plugin depends on some pip module (such as the cudatext ftp plugin dependency on paramiko) and that pip module is not in the flatpak python installation that the io.github.cudatext.CudaText-Qt5 package uses.

**Comment by bogen85** [comment]
I opened up issue #3 .

Since the python engine is working with CudaText for the flatpak install of the io.github.cudatext.CudaText-Qt5 package. This issue can be closed. (As the Python engine is already working with the flatpak install).

**Comment by bogen85** [comment]
From `user.json`
```json
  "ui_theme" : "white-qt5-fix (dark)",
  "ui_theme_syntax" : "cobalt",
  "pylib__linux" : "libpython3.so",
  "font_name__linux" : "Fira Code Retina",
  "font_size__linux" : 16,
```
that results in:
```
Python 3.9.9
Init: cuda_project_man
Loading project: ~/.var/app/io.github.cudatext.CudaText-Qt5/config/cudatext/settings/temporary.cuda-proj
...
Loaded session: "history session.json", 10ms, 2 file(s)
Startup: 460ms, plugins: 60ms (project_man 48ms, modeline 13ms)
```
The CudaText python engine is working fine with the flatpak install of CudaText.

**Comment by bogen85** [comment]
@Alexey-T the full python path would be one of these locations...

```
$ locate libpython3.so | fgrep flatpak
/var/lib/flatpak/.removed/org.kde.Platform-3f92e49eb0d2f4b48b3b19537433316a331a5bcb39a6c300664725c0ce2a1fa9/files/lib/x86_64-linux-gnu/libpython3.so
/var/lib/flatpak/runtime/org.freedesktop.Platform/x86_64/21.08/c7252386179c4c1ecf5d93bdaec6ca82852dddfdae7112b5c3177f7424c4a928/files/lib/x86_64-linux-gnu/libpython3.so
/var/lib/flatpak/runtime/org.freedesktop.Platform/x86_64/22.08/289b339fd2050a03e90dd05b937b7b2320ad343eb8389a6e1eb6e87cfdb88e1e/files/lib/x86_64-linux-gnu/libpython3.so
/var/lib/flatpak/runtime/org.gnome.Platform/x86_64/3.38/454a79938e3ecb6ee3750055b1cfb2d28435b0ffe1ff5742b092917cf7efc8a1/files/lib/x86_64-linux-gnu/libpython3.so
/var/lib/flatpak/runtime/org.gnome.Platform/x86_64/41/a6d16f911222999f9a11e37580ca42abb350b143929381f4b781b4917f3cfb9e/files/lib/x86_64-linux-gnu/libpython3.so
/var/lib/flatpak/runtime/org.gnome.Platform/x86_64/42/e461129c14120dcf8801bf1bb25257a651cee3f2fd0ebd1a31c1d51db1da9e9c/files/lib/x86_64-linux-gnu/libpython3.so
/var/lib/flatpak/runtime/org.kde.Platform/x86_64/5.15-21.08/8184db77dbe46028a5efc42980d07b6d91c4dcf853be7cd5d3d98d9893ca297d/files/lib/x86_64-linux-gnu/libpython3.so
```

**Comment by hfiguiere** [comment]
not it's not. You are just viewing the ostree storage.

A fatpak app run in a sandbox.

**Comment by bogen85** [comment]
@hfiguiere , you are correct.
```
$ ls /proc/$(pidof cudatext)/map_files/ -ltr | fgrep libpython3.so | sed "s/$USER/USER/g"
lr-------- 1 USER USER 64 Sep 10 08:58 6c346c243000-6c346c244000 -> /usr/lib/x86_64-linux-gnu/libpython3.so
lr-------- 1 USER USER 64 Sep 10 08:58 6c346c242000-6c346c243000 -> /usr/lib/x86_64-linux-gnu/libpython3.so
lr-------- 1 USER USER 64 Sep 10 08:58 6c346c241000-6c346c242000 -> /usr/lib/x86_64-linux-gnu/libpython3.so
lr-------- 1 USER USER 64 Sep 10 08:58 6c346c240000-6c346c241000 -> /usr/lib/x86_64-linux-gnu/libpython3.so
lr-------- 1 USER USER 64 Sep 10 08:58 6c346c23f000-6c346c240000 -> /usr/lib/x86_64-linux-gnu/libpython3.so
```

**Comment by hfiguiere** [comment]
tl;dr it's the standard location (on x86_64, aarch64 is left as an exercise to the reader)
